core/commands/pin/remotepin.go

```diff
@@ -89,20 +89,41 @@ func printRemotePinDetails(w io.Writer, out *RemotePinOutput) {
 
 // remote pin commands
 
-var pinServiceNameOption = cmds.StringOption(pinServiceNameOptionName, "Name of the remote pinning service to use.")
+var pinServiceNameOption = cmds.StringOption(pinServiceNameOptionName, "Name of the remote pinning service to use (mandatory).")
 
 var addRemotePinCmd = &cmds.Command{
 	Helptext: cmds.HelpText{
 		Tagline:          "Pin object to remote pinning service.",
-		ShortDescription: "Stores an IPFS object from a given path to a remote pinning service.",
+		ShortDescription: "Asks remote pinning service to pin an IPFS object from a given path.",
+		LongDescription: `
+Asks remote pinning service to pin an IPFS object from a given path or a CID.
+
+To pin CID 'bafkqaaa' to service named 'mysrv' under a pin named 'mypin':
+
+  $ ipfs pin remote add --service=mysrv --name=mypin bafkqaaa
+
+The above command will block until remote service returns 'pinned' status,
+which may take time depending on the size and available providers of the pinned
+data.
+
+If you prefer to not wait for pinning confirmation and return immediately
+after remote service confirms 'queued' status, add the '--background' flag:
+
+  $ ipfs pin remote add --service=mysrv --name=mypin --background bafkqaaa
+
+Status of background pin requests can be inspected with the 'ls' command:
+
+  $ ipfs pin remote ls --service=mysrv --cid=bafkqaaa --status=queued,pinning,pinned,failed
+
+`,
 	},
 
 	Arguments: []cmds.Argument{
-		cmds.StringArg("ipfs-path", true, false, "Path to object(s) to be pinned."),
+		cmds.StringArg("ipfs-path-or-cid", true, false, "Path to object(s) to be pinned."),
```

Comment:
I think leaving this at `ipfs-path` is actually less confusing since this is what we call it in other functions (e.g. `ipfs get`). It also corresponds to the paths in https://github.com/ipfs/go-path.